aiida/common/setup.py

```diff
@@ -136,73 +136,6 @@ def store_config(confs):
         os.umask(old_umask)
 
 
-def install_daemon_files(aiida_dir, daemon_dir, log_dir, local_user,
-                         daemon_conf=None):
-    """
-    Install the files needed to run the daemon.
-    """
-    local_daemon_conf = """
-[unix_http_server]
-file={daemon_dir}/supervisord.sock   ; (the path to the socket file)
-
-[supervisord]
-logfile={log_dir}/supervisord.log
-logfile_maxbytes=10MB
-logfile_backups=2
-loglevel=info
-pidfile={daemon_dir}/supervisord.pid
-nodaemon=false
-minfds=1024
-minprocs=200
-
-[rpcinterface:supervisor]
-supervisor.rpcinterface_factory = supervisor.rpcinterface:make_main_rpcinterface
-
-[supervisorctl]
-serverurl=unix:///{daemon_dir}/supervisord.sock
-
-;=======================================
-; Main AiiDA Daemon
-;=======================================
-[program:aiida-daemon]
-command=celery worker -A tasks --loglevel=INFO --beat --schedule={daemon_dir}/celerybeat-schedule
-directory={aiida_code_home}/daemon/
-user={local_user}
-numprocs=1
```

Comment:
I think this is the line that was limiting the # of procs to 1 (and so probably indirectly the # of celery workers).

Reply:
No, it does not. It only tells supervisor to start only one `celery worker` process instead of multiple ones.

Here is the the analysis with the current develop branch and using supervisor:

```
(venv) tiziano@tcpc18 ~/work/aiida/aiida_core (develop *=) $ verdi daemon start
11/04/2017 02:45:19 PM, alembic.runtime.migration: [INFO] Context impl PostgresqlImpl.
11/04/2017 02:45:19 PM, alembic.runtime.migration: [INFO] Will assume transactional DDL.
Clearing all locks ...
Starting AiiDA Daemon ...
Daemon started
(venv) tiziano@tcpc18 ~/work/aiida/aiida_core (develop *=) $ verdi daemon status
11/04/2017 02:48:26 PM, alembic.runtime.migration: [INFO] Context impl PostgresqlImpl.
11/04/2017 02:48:26 PM, alembic.runtime.migration: [INFO] Will assume transactional DDL.
# Most recent daemon timestamp:0h:00m:02s ago
## Found 1 process running:
   * aiida-daemon[aiida-daemon] RUNNING    pid 17019, uptime 0:03:05
(venv) tiziano@tcpc18 ~/work/aiida/aiida_core (develop *=) $ pstree -p 17019
celery(17019)─┬─celery(17033)
              ├─celery(17034)
              ├─celery(17035)
              ├─celery(17036)
              ├─celery(17037)
              ├─celery(17038)
              ├─celery(17039)
              ├─celery(17040)
              ├─celery(17044)
              ├─{celery}(17041)
              ├─{celery}(17042)
              └─{celery}(17043)
```

and here is the output without supervisor:

```
(venv) tiziano@tcpc18 ~/work/aiida/aiida_core (supervisor-removal *>) $ verdi daemon start
11/04/2017 02:51:59 PM, alembic.runtime.migration: [INFO] Context impl PostgresqlImpl.
11/04/2017 02:51:59 PM, alembic.runtime.migration: [INFO] Will assume transactional DDL.
Clearing all locks ...
Starting AiiDA Daemon (log file: /users/tiziano/.aiida/daemon/log/celery.log)...
Daemon started
(venv) tiziano@tcpc18 ~/work/aiida/aiida_core (supervisor-removal *>) $ verdi daemon status
11/04/2017 02:52:13 PM, alembic.runtime.migration: [INFO] Context impl PostgresqlImpl.
11/04/2017 02:52:13 PM, alembic.runtime.migration: [INFO] Will assume transactional DDL.
# Most recent daemon timestamp:0h:00m:01s ago
Daemon is running as pid 17868 since 2017-11-04 14:51:59.590000, child processes:
   * celery[17899]   sleeping, started at 2017-11-04 14:52:01
   * celery[17900]   sleeping, started at 2017-11-04 14:52:01
   * celery[17901]   sleeping, started at 2017-11-04 14:52:01
   * celery[17902]   sleeping, started at 2017-11-04 14:52:01
   * celery[17903]   sleeping, started at 2017-11-04 14:52:01
   * celery[17904]   sleeping, started at 2017-11-04 14:52:01
   * celery[17905]   sleeping, started at 2017-11-04 14:52:01
   * celery[17906]   sleeping, started at 2017-11-04 14:52:01
   * celery[17910]   sleeping, started at 2017-11-04 14:52:01
(venv) tiziano@tcpc18 ~/work/aiida/aiida_core (supervisor-removal *>) $ pstree -p 17868
celery(17868)─┬─celery(17899)
              ├─celery(17900)
              ├─celery(17901)
              ├─celery(17902)
              ├─celery(17903)
              ├─celery(17904)
              ├─celery(17905)
              ├─celery(17906)
              ├─celery(17910)
              ├─{celery}(17907)
              ├─{celery}(17908)
              └─{celery}(17909)
```

I used `verdi setup ...` using the SQLA backend to setup my profile, so I don't think there is something wrong with it.

Can you please check on one of your setups (preferably on one with more than 1 CPU).